rust/noosphere/src/ffi/noosphere.rs

```diff
@@ -0,0 +1,74 @@
+use anyhow::Result;
+use safer_ffi::prelude::*;
+use url::Url;
+
+use crate::noosphere::{NoosphereContext as NoosphereContextImpl, NoosphereContextConfiguration};
+
+ReprC! {
+    #[ReprC::opaque]
+    pub struct NoosphereContext {
+        inner: NoosphereContextImpl
```

Comment:
Multiple e.g. `NoosphereContext` types seem like it could get confusing even with restricting the FFI interfaces to `noosphere::ffi::*` -- though I don't feel strongly, have no (good) alternatives, and the name collision is relatively contained.